package/ios/RNSkia-iOS/RNSkiOSPlatformContext.mm

```diff
@@ -69,7 +152,32 @@
 sk_sp<SkImage>
 RNSkiOSPlatformContext::makeImageFromPlatformBuffer(void *buffer) {
   CMSampleBufferRef sampleBuffer = (CMSampleBufferRef)buffer;
-  return SkiaMetalSurfaceFactory::makeImageFromCMSampleBuffer(sampleBuffer);
+  // Step 1: Extract the CVPixelBufferRef from the CMSampleBufferRef
+  CVPixelBufferRef pixelBuffer = CMSampleBufferGetImageBuffer(sampleBuffer);
+
+  // Step 2: Lock the pixel buffer to access the raw pixel data
+  CVPixelBufferLockBaseAddress(pixelBuffer, 0);
+
+  // Step 3: Get information about the image
+  void *baseAddress = CVPixelBufferGetBaseAddress(pixelBuffer);
+  size_t width = CVPixelBufferGetWidth(pixelBuffer);
+  size_t height = CVPixelBufferGetHeight(pixelBuffer);
+  size_t bytesPerRow = CVPixelBufferGetBytesPerRow(pixelBuffer);
+
+  // Assuming the pixel format is 32BGRA, which is common for iOS video frames.
+  // You might need to adjust this based on the actual pixel format.
+  SkImageInfo info = SkImageInfo::Make(width, height, kRGBA_8888_SkColorType,
+                                       kUnpremul_SkAlphaType);
+
+  // Step 4: Create an SkImage from the pixel buffer
+  sk_sp<SkData> data = SkData::MakeWithCopy(baseAddress, height * bytesPerRow);
```

Comment:
No, accessing the CVPixelBuffer like this is going to be really slow and inefficient since your code performs two CPU copies:

1. `CVPixelBufferLockBaseAddress` - the IOSurface is usually on the GPU, but this creates a CPU handle
2. `MakeWithCopy` - this does a CPU copy in our own memory. 

For 4k buffers streaming in at 60 or 120 FPS, this is going to be too slow.
The approach I used uses Metal and a GPU surface, so it does not do any CPU copies and is suitable for realtime use (and recommended by Apple and other guides)

Reply:
I don't want this get too messy because of multiple PRs - but just as a sanity check, can you test if my new PR (https://github.com/Shopify/react-native-skia/pull/2354) works for you with RGB? I changed one tiny thing about RGB buffers in there.

Reply:
yes if we merge this PR we can run the e2e test on #2354 directly. This particular change is only to get the test passing on iOS